go/deaggregator/deaggregator.go

```diff
@@ -22,9 +22,9 @@ const (
 
 // DeaggregateRecords takes an array of Kinesis records and expands any Protobuf
 // records within that array, returning an array of all records
-func DeaggregateRecords(records []*kinesis.Record) ([]*kinesis.Record, error) {
+func DeaggregateRecords(records []*types.Record) ([]*types.Record, error) {
```

Comment:
thoughts on updating this to use `[]types.Record`? The v2-sdk doesn't use pointers which means any consumer of this library is going to need to jump through some hoops to create a new slice of pointers to pass to this func

```
// Represents the output for GetRecords.
type GetRecordsOutput struct {

	// The data records retrieved from the shard.
	//
	// This member is required.
	Records []types.Record
```